File: SocialMedia_Main/ClientApp/src/app/Interfaces/chat-head-view-model.ts
import { UserViewModel } from "./user-view-model";

export interface ChatHeadViewModel {
  receiverId: string;
  lastMessage: string;
  isMessageRead: boolean;
  receiverFirstName: string;
  receiverLastName: string;
  chatHeadId: string;
  user: UserViewModel
}

export interface AddMessageViewModel {
  chatId: string;
  senderId: string;
  receiverId: string;
  chatHeadId: string;
  message: string;
}

export interface  MessageResponse {
  chatId: string;
  senderId: string;
  senderUser: any;
  receiverId: string;
  receiverUser: any;
  message: string;
  sentTime: string;
  isMessageRead: boolean;
  chatHead: ChatHeadViewModel;
  chatHeadId: string;
}

File: SocialMedia_Common/ChatView/ChatHeadViewModel.cs
using SocialMedia_Data.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SocialMedia_Common.ChatView
{
    public class ChatHeadViewModel
    {
        public string ReceiverId { get; set; }
        public string LastMessage { get; set; }
        public bool IsMessageRead { get; set; }
        public string ReceiverFirstName { get; set; }
        public string ReceiverLastName { get; set; }
        public string ChatHeadId { get; set; }
        public User User { get; set; }
    }
}
